src/axom/quest/DistributedClosestPoint.hpp

```diff
@@ -680,207 +680,197 @@ class DistributedClosestPointImpl
+    {
+    case RuntimePolicy::seq:
+      computeLocalClosestPoints<SeqBVHTree>(m_bvh_seq.get(), xferNode);
+      break;
+
+    case RuntimePolicy::omp:
 #ifdef _AXOM_DCP_USE_OPENMP
-          computeLocalClosestPoints<OmpBVHTree>(m_bvh_omp.get(), xferNode);
+      computeLocalClosestPoints<OmpBVHTree>(m_bvh_omp.get(), xferNode);
 #endif
-          break;
+      break;
 
-        case RuntimePolicy::cuda:
+    case RuntimePolicy::cuda:
 #ifdef _AXOM_DCP_USE_CUDA
-          computeLocalClosestPoints<CudaBVHTree>(m_bvh_cuda.get(), xferNode);
+      computeLocalClosestPoints<CudaBVHTree>(m_bvh_cuda.get(), xferNode);
 #endif
-          break;
+      break;
 
-        case RuntimePolicy::hip:
+    case RuntimePolicy::hip:
 #ifdef _AXOM_DCP_USE_HIP
-          computeLocalClosestPoints<HipBVHTree>(m_bvh_hip.get(), xferNode);
+      computeLocalClosestPoints<HipBVHTree>(m_bvh_hip.get(), xferNode);
 #endif
-          break;
-        }
-
-        if(xferNode.fetch_existing("homeRank").as_int() == m_rank)
-        {
-          // This is the query partition we started with, completing its trip
-          // around the ring.  Copy it back to queryMesh, if necessary
-          const bool should_copy = xferNode.has_child("qPtCount");
-          if(should_copy)
-          {
-            copy_xfer_node_to_query_node(xferNode, queryMesh);
-
-            if(m_isVerbose)
-            {
-              internal::dump_node(
-                queryMesh,
-                axom::fmt::format("round_{}_r{}_end.json", round, m_rank));
-
-              SLIC_ASSERT_MSG(
-                conduit::blueprint::mcarray::is_interleaved(
-                  queryMesh["fields/closest_point/values"]),
-                fmt::format(
-                  "After copy on iteration {}, 'closest_point' field of "
-                  "'queryMesh' is not interleaved",
-                  round));
-            }
-          }
-          xferNodePtr.reset();
-        }
-      }  // Locally process *xferNodePtr
-
-      SLIC_INFO_IF(
-        m_isVerbose,
-        fmt::format("=======  End of round {}/{} =======", round, m_nranks));
-    }  // round loop
+      break;
+    }
+  }
 
-    // Complete non-blocking sends.
-    while(!isendRequests.empty())
+  /**
+    Determine the next rank (in ring order) with an object partition
+    close to the query points in xferNode.  The intent is to send
+    xferNode there next.
+  */
+  int next_recipient(const conduit::Node& xferNode) const
+  {
+    int homeRank = xferNode.fetch_existing("homeRank").value();
+    BoxType bb;
+    get_bounding_box_from_conduit_node(bb, xferNode.fetch_existing("aabb"));
+    for(int i = 1; i < m_nranks; ++i)
     {
-      check_send_requests(isendRequests, true);
+      int maybeNextRecip = (m_rank + i) % m_nranks;
+      if(maybeNextRecip == homeRank)
+      {
+        return maybeNextRecip;
+      }
+      double sqDistance =
+        primal::squared_distance(bb, m_objectPartitionBbs[maybeNextRecip]);
```

Comment:
... as a follow up to discussions in other threads -- this check for `maybeNextRecip` only accounts for the distance between the query mesh's bounding box and the object meshes bounding box (with respect to the distance threshold).

Perhaps it can also account for the approximate max distances that we've already computed.
E.g. we might already know that the max distance from a query point on this rank to its closest point is at most `_conservative_max`. Then, if the max distance from `bb` to `m_objectPartitionBbs[maybeNextRecip]` is more than this, we can skip rank `maybeNextRecip`.

Since this is an optimization, and it can get messy with the bookkeeping related to the number of ranks to check, we might want to think about this for future PRs.

Reply:
I get the gist of this, and I agree.